ingest/src/test/java/com/microsoft/azure/kusto/ingest/ManagedStreamingIngestClientTest.java

```diff
@@ -0,0 +1,352 @@
+
+        verifyCompressedStreamContent(argumentCaptor.getValue(), uncompressedContents);
+    }
+
+    @Test
+    void IngestFromStream_Success() throws Exception {
+        String data = "Name, Age, Weight, Height";
+        InputStream inputStream = new ByteArrayInputStream(StandardCharsets.UTF_8.encode(data).array());
+        StreamSourceInfo streamSourceInfo = new StreamSourceInfo(inputStream);
+        OperationStatus status = managedStreamingIngestClient.ingestFromStream(streamSourceInfo, ingestionProperties).getIngestionStatusCollection().get(0).status;
+        assertEquals(status, OperationStatus.Succeeded);
+        verify(streamingClientMock, atLeastOnce()).executeStreamingIngest(any(String.class), any(String.class), argumentCaptor.capture(),
+                isNull(), any(String.class), eq("mappingName"), any(boolean.class));
+
+        /* In order to make efficient ingestion requests, the streaming ingest client compress the given stream unless it is already compressed.
+         * When the given stream content is already compressed, the user must specify that in the stream source info.
+         * This method verifies if the stream was compressed correctly.
+         */
+        InputStream stream = argumentCaptor.getValue();
+        verifyCompressedStreamContent(stream, data);
+    }
+
+    @Test
+    void IngestFromStream_FailStreaming() throws Exception {
+        try {
+            final int[] times = {0}; //it's an array so we can safely modify it in the lambda
+            String data = "Name, Age, Weight, Height";
+            InputStream inputStream = new ByteArrayInputStream(StandardCharsets.UTF_8.encode(data).array());
+            StreamSourceInfo streamSourceInfo = new StreamSourceInfo(inputStream);
+
+            when(streamingClientMock.executeStreamingIngest(any(String.class), any(String.class), argumentCaptor.capture(),
+                    isNull(), any(String.class), eq("mappingName"), any(boolean.class)))
+                    .thenAnswer((a) -> {
+                        times[0]++;
+                        throw new DataServiceException("Test fail");
+                    });
+
+            //should fail 10 times and then succeed with the queued client
+            managedStreamingIngestClient.ingestFromStream(streamSourceInfo, ingestionProperties);
+            assertEquals(ManagedStreamingIngestClient.MAX_RETRY_CALLS, times[0]);
+
+
+        } finally {
+            //reset the mock
+            when(streamingClientMock.executeStreamingIngest(any(String.class), any(String.class), argumentCaptor.capture(),
+                    isNull(), any(String.class), eq("mappingName"), any(boolean.class)))
+                    .thenReturn(null);
+        }
+    }
+
+    @Test
+    void IngestFromStream_FailFewTimes() throws Exception {
+        int failCount = 2;
+        final int[] times = {0}; //it's an array so we can safely modify it in the lambda
```

Comment:
Here and all comments: we actually comment here like in the .Net code- move comments above the line, space after the `//` and first letter is capital letter . Actually I don't think you need to explain these stuff on comments ..

Reply:
Do we have a style guide somewhere?

And as for the explanation, IDK if I refactored this code someday I might think "why isn't this an int" and forget about this 'gotcha', so I think this comment is nice.